src/db.ts

```diff
@@ -324,6 +324,7 @@ export class Db {
     if (!options) {
       options = {};
     } else if (typeof options === 'function') {
+      // TODO: Replace this error with a more appropriate error class
```

Comment:
Considering making this a MongoCompatibilityError

Reply:
I think it's good to only use MongoCompatiblityError when the server doesn't support a given feature. I'd leave this as is or maybe introduce a MongoDeprecationError that is thrown when a user tries to use a previously deprecated functionality that is no longer supported. This error is only here as a convenience to users who are upgrading, so my vote would be to just leave as is, maybe with a note to remove it entirely in driver 5.0.